### PR TITLE
Task/ctaa 1665  fix questionnaire

### PR DESCRIPTION
**Pull request by JU5T1C3**
## Changes

Fixed doubled questions in the selftest.

## Solved issues

- [Issue #1665](https://tasks.pxp-x.com/browse/CTAA-1665)

### CODE REVIEW

File: app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireViewModel.kt [hunk 1]
```diff
@@ -106,9 +107,13 @@ class QuestionnaireViewModel(
...
-    fun observeFetchState(): Observable<State> = questionnaireStateObserver.observe()
+    fun observeFetchState(): Observable<State> = fetchQuestionnaireStateObserver.observe()
 
-    fun observeQuestionnaireWithAnswers(language: ConfigurationLanguage): Observable<List<DbQuestionnaireWithAnswers>> {
-        return configurationRepository.observeQuestionnaireWithAnswers(language)
+    fun observeQuestionnaireWithQuestions(): Observable<List<DbQuestionnaireWithAnswers>> = questionnaireSubject
+
+    fun getQuestionnaireWithAnswers(language: ConfigurationLanguage) {
+        launch {
+            questionnaireSubject.onNext(configurationRepository.getQuestionnaireWithAnswers(language))
```
Comment (mihbt) [inline app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireViewModel.kt]: This call could trigger an exception that is not handled. We can move the logic of getting the questionnaire with answers inside the `fetchQuestionnaire` method

File: app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireFragment.kt [hunk 2]
```diff
@@ -133,6 +124,17 @@ class QuestionnaireFragment : BaseFragment(R.layout.fragment_questionnaire) {
                     Decision.SELFMONITORING -> startQuestionnaireSelfMonitoringFragment()
                 }
             }
+
+        viewModel.getQuestionnaireWithAnswers(
```
Comment (dusanjencik) [inline app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireFragment.kt]: If you are not observing new data anymore, the fetch on start is useless.

**Review by dusanjencik — CHANGES_REQUESTED** [review]
Nice

File: app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireViewModel.kt (hunk 1 above)
Comment (JU5T1C3) [reply app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireViewModel.kt]: done

File: app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireFragment.kt (hunk 2 above)
Comment (JU5T1C3) [reply app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireFragment.kt]: done

**Comment by mihbt** [comment]
Nicely done!

**Review by mihbt — APPROVED** [review]

**Review by dusanjencik — APPROVED** [review]

**Pull request merged**